### ISSUE TITLE
Swimlane feature messes with display of detailed view

### ISSUE THREAD

**Issue by Badpat** [wekan/wekan#1444]
## Issue
**Impacted version**: 0.67.0

**Server Setup Information**:

* Operating System: Ubuntu
* Deployment Method: docker with mongodb using https://github.com/wekan/wekan-mongodb
* Node Version: v8.9.3
* MongoDB Version: 3.2.17
* ROOT_URL environment variable (Is there a subfolder?): Yes (https://domain/subfolder/wekan)

**Problem description**:
- *be as explicit has you can*
- *describe the problem and its symptoms*
- *explain how to reproduce*
- *attach whatever information that can help understanding the context (screen capture, log files)*

Today I upgraded to the latest wekan version. After upgrade I now have the swimlane feature but this leads to some problems:

1) If I deactivate the swimlane view, cards are not being opened correctly. The mini-card moves slightly to the right, as usual, but there is no detailed view being displayed.

2) If swimlane view is used, the detailed view of the opened card is being shown multiple times. Once in each lane.

Tested with Chrome (Version 64.0.3282.119) and Firefox (Version 58.0).

**Comment by johnleeming** [comment]
Swimlanes also seem to mess with drag & drop ordering of cards in a list - if I move a card in swimlanes display it juts goes back to where it was.  In the list view dragging on a card moves it out of the window and puts a whole bunch of placeholders in eth list but won't allow teh card back in.

Wekan 0,67 on Heroku, browser is Vivaldi | 1.11.917.43

**Comment by fbatschi** [comment]
Further details for case 1:
The JS Console shows an error "There is no route for the path /b/boardid/boardname/cardname whith boardname and cardname being the corresponding values from the board.

**Comment by fbatschi** [comment]
is there a problem to downgrade? Did anyone try this?
Or does this mess with the database structure?

The current version is actually unusable with the swimlane feature.

**Comment by xet7** [comment]
@fbatschi 

I do recommend backups anyway before upgrade and downgrade, just in case.
https://github.com/wekan/wekan/wiki/Export-Docker-Mongo-Data

Example of downgrade:
https://github.com/wekan/wekan/issues/1438#issuecomment-361075193

@andresmanelli 

Can you look at above comments for future swimlanes improvements?

**Comment by DissCent** [comment]
I experience the same problems - but for some reason, I also can't switch from swimlanes mode to the classic one - if I try, the Javascript console reports something like "Access denied" and the UI returns back to the swimlanes view.

**Comment by xet7** [comment]
@DissCent 

Do you mean bugs that recent Chrome version did bring when they broke stuff?
https://github.com/wekan/wekan/issues/656

For me Wekan works fine in Firefox. Chrome is just slow, memory-hungry and buggy, I only use it when something does not work in Firefox.

**Comment by DissCent** [comment]
@xet7 I use Firefox as my main browser :) but I also can't move cards in the Swimlane view and have the issue that I can't exit the swimlane view.

**Comment by xet7** [comment]
@DissCent 

I presume you have old version of Wekan. I can move cards and switch lists/swimlane view just fine in newest version. Please reinstall using some of these tags:

```
image: quay.io/wekan/wekan
or
image: wekanteam/wekan:v0.63
```

Here is info of how to upgrade:
https://github.com/wekan/wekan/issues/1438#issuecomment-361075193

**Comment by xet7** [comment]
@DissCent 

What version of Firefox you have? I have Firefox v58 for Ubuntu 17.10 64bit.

What operating system you use? Linux/Windows/Mac/something else ?

**Comment by DissCent** [comment]
@xet7 I also use Firefox 58.0 on Manjaro Linux (which is basically the same as Arch Linux) x64.

**Comment by xet7** [comment]
@DissCent 

Can you try this image?

```
image: quay.io/wekan/wekan
```

I don't really know where you are getting that bug. I do see that there is build failures at Docker Hub. Quay usually works better.

**Comment by DissCent** [comment]
@xet7 I already tried that one a couple of minutes ago - but still no luck :(

**Comment by xet7** [comment]
@DissCent 

In Admin Panel/Version does it show that Wekan version on v0.68 ? If not, you have not removed old containers and images, and you are running old version.

**Comment by xet7** [comment]
It seems that Docker Hub has old version.

**Comment by DissCent** [comment]
@xet7 

So I stopped wekan-app, did "docker rm wekan-app" and removed all wekan-app images with "docker rmi <...>". I rolled back my database to v0.63 format, set the image to "quay.io/wekan/wekan" and used "docker-compose up -d". The version inside Wekan now shows "0.68.0". However, the issues still remain (also tested with other browsers).

As an example, if I press the button "Swimlanes" to switch the view, the following gets sent to the server:

`{"msg":"method","method":"/boards/update","params":[{"_id":"NxTfQAHsCa6CDGKuC"},{"$set":{"view":"board-view-lists","modifiedAt":{"$date":1517315412066}}},{}],"id":"13"}`

And this is the response:

`a["{\"msg\":\"result\",\"id\":\"13\",\"error\":{\"isClientSafe\":true,\"error\":403,\"reason\":\"Access denied\",\"message\":\"Access denied [403]\",\"errorType\":\"Meteor.Error\"}}"]`

**Comment by xet7** [comment]
@DissCent 

I think your MongoDB install is broken.

You need to backup your MongoDB database with mongodump, reinstall MongoDB like for example remove MongoDB container and image, and restore:
https://github.com/wekan/wekan/wiki/Export-Docker-Mongo-Data

You can use MongoDB version that's in Docker Compose file:
https://github.com/wekan/wekan-mongodb

**Comment by johnleeming** [comment]
I have updated to 0.68 - pull request to my Github clone then Heroku autobuilds from that.  I am still getting the errors
![image](https://user-images.githubusercontent.com/13801070/35568482-11c2eb94-05c1-11e8-98d9-90aebf09a1b8.png).

I try to move "job 5" in this test board in list view and get
![image](https://user-images.githubusercontent.com/13801070/35568530-3c99e020-05c1-11e8-837b-4260a37a3b82.png)
in swim lane view I can move cards between lanes but if I try to change order of cards within one swim lane it just goes back to where it was before

**Comment by DissCent** [comment]
@xet7 

I cleaned my MongoDB container and with a clean install, switching between swimlanes and lists now works correctly.
However, the same issues as @johnleeming still remain for me.

**Comment by andresmanelli** [comment]
@DissCent @johnleeming I'll try to take a look this week, I have the same behavior.

**Comment by andresmanelli** [comment]
Update:

Now details view is displayed in lists-only mode. Could someone give it a try?

**Comment by gramakri** [comment]
@andresmanelli I tried your branch quickly but getting some errors (I am using ff9ca755f338e3c45a1bd726dfbce1c607f2ff4c from your fork). This is a fresh installation and clicking 'Add' does not work.

```
There is no route for the path: / f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:5870
	router.js/f.prototype._notfoundRoute https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:5870
	router.js/f.prototype._updateCallbacks/< https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:7528
	index.js/s.prototype.middleware/< https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:15549
	r https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:14952
	index.js/n.dispatch https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:15004
	index.js/n.replace https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:14783
	router.js/f.prototype.initialize/</e._page[t] https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:6212
	index.js/n.start https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:14080
	n https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:11779
	router.js/f.prototype.initialize https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:6268
	_init.js/< https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:247:11285
	a https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:1:5282
	u https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:1:5370
```

```
TypeError: Swimlanes.findOne(...) is undefined[Learn More] f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:289:153002
	addCard https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:289:153002
	u/t[n]/</< https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:217:6103
	f._withCurrentView https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:59:23097
	u/t[n]/< https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:217:6064
	e._withTemplateInstanceFunc https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:59:37698
	u/t[n] https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:217:6042
	i/</</</< https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:59:27013
	f._withCurrentView https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:59:23097
	i/</</< https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:59:26974
	a/this.delegatedHandler</< https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:59:9120
	dispatch https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:19:49877
	add/g.handle https://test.smartserver.space/f361e2be3bd4e39c74d7f68b56d033ddce131baf.js:19:46580
```

**Comment by andresmanelli** [comment]
@gramakri  I am getting myself errors of the kind `There is no route [...]`. Even when I load the main page.

For me, reloading a board gives errors, but going to the board _from the home site_ works fine. This started happening after the Meteor update, I do not know if it is related.

**Comment by nebulade** [comment]
I have pushed a pull request https://github.com/wekan/wekan/pull/1454 to fix the list view card moving. The details view is still not showing with this though, but at least the issue from @johnleeming showed in the screenshot should be gone.

**Comment by nebulade** [comment]
I am not sure how the relationship in the ui is between the swimlanes view and the lists view, but swimlanes appears to be based off the lists view, however the lists view as it currently is, does not appear to have any reference to show the card details panel. Maybe someone with more knowledge here could explain where the lists view would ever show the details panel, since it misses the `+cardDetails(currentCard)` logic from the swimlanes.

**Comment by andresmanelli** [comment]
@nebulade could you take a look at the last 3 commits of my fork? (See comments above). These bugs are fixed for me (maybe more tests are necessary) including the one fixed by your PR.

I _think_ your commit maybe breaks the drag in swimlane view.

Sorry if maybe I'm mistaken, I'll give it a look later.

**Comment by nebulade** [comment]
@andresmanelli ah nice, indeed I didn't look at all at your fork and was about to add similar changes for me, however given my poor code base knowledge as well as meteor knowledge, mine are just hacks ;-)

So I can confirm the three commits in your fork are fixing all those issues without side-effects I can make out. I think I will cherry-pick them for our Cloudron package to have a working app for our users. Thanks a lot, hope they get into mainline soon.

**Comment by xet7** [comment]
Fixed in Wekan v0.69, I think

**Comment by johnleeming** [comment]
seems to work fine for me - thanks very much.

I can change the order of cards in both list and swimlane views and also open card details in both views
